### PR TITLE
Implemented standalone Settings page

### LINKED ISSUE #185: Finish Settings page implementation
# Overview

Two tabs: Github / Amazon S3
- GitHub
  - A button to sync the user account (ie get all user’s repos)
  - A list of repos, with the ones active on top, and a switch to activate or deactivate them
  - Ideally the repos would be filtered by organization to help navigate them but 
- Amazon S3
  - A list of active buckets, with name, link to jobs and option to remove / deactivate
  - A form to add new buckets, with three fields: name, access key id and secret access

# Tasks

- [ ] TBD

### PR DESCRIPTION
**Pull request by roll**
- fixes #185

---

- [ ] add design issues to #187

### CODE REVIEW

**Comment by roll** [comment]
@amercader 
I had a real strange bug on clicking `remove s3 bucket` (my user session breaks) - could you check?

**Comment by amercader** [comment]
@roll maybe this:

```diff
diff --git a/goodtablesio/utils/frontend.py b/goodtablesio/utils/frontend.py
index 20d2987..c5c3366 100644
--- a/goodtablesio/utils/frontend.py
+++ b/goodtablesio/utils/frontend.py
@@ -22,7 +22,7 @@ def render_component(component, props={}):
         filename = 'index.html'
 
     # Common props
-    if not props or (props and 'userName' not in props):
+    if not props or props == {} or (props and 'userName' not in props):
         props['userName'] = getattr(current_user, 'display_name', None)
 
     return render_template(filename, component=component, props=props.copy())

```

It worked fine for me after it

**Comment by amercader** [comment]
you are right, something is wrong with sessions. When I tried to go to the dashboard I was logged out. I'll investigate a bit more.

File: goodtablesio/blueprints/site.py [hunk 1]
```diff
@@ -64,6 +64,11 @@ def source_s3_job(bucket, job):
     return _source('s3', bucket, job)
 
 
+@site.route('/settings')
+def settings():
```
Comment (amercader) [inline goodtablesio/blueprints/site.py]: You'll need `@login_required` here (just noticed as I'm working on access checks)

**Pull request merged**